### ISSUE TITLE
:matches() not parsing correctly

### ISSUE THREAD

**Issue by ianstormtaylor** [threepointone/glamor#164]
I have a definition like:

```js
css({
  '& :matches(h1, h2, h3, h4, h5, h6)': {
    marginTop: '2.5em',
    marginBottom: '.8em',
  }
})
```

In attempt to match child heading elements. Expecting the generated CSS to be:

```css
.css-owi01 :matches(h1, h2, h3, h4, h5, h6),[data-css-owi01] :matches(h1, h2, h3, h4, h5, h6){margin-top:2.5em;margin-bottom:.8em;}
```

But instead it is:

```css
.css-owi01 :matches(h1,[data-css-owi01] :matches(h1,.css-owi01 h2,[data-css-owi01] h2,.css-owi01 h3,[data-css-owi01] h3,.css-owi01 h4,[data-css-owi01] h4,.css-owi01 h5,[data-css-owi01] h5,.css-owi01 h6),[data-css-owi01] h6){margin-top:2.5em;margin-bottom:.8em;}
```

Am I doing something dumb?

(It's currently supported in most browsers without the need for the expand-into-combinations fallback that used to be performed by pre-processors, so I think it would be easiest to just treat it as a regular pseudo-selector and pass it straight through.)

**Comment by threepointone** [comment]
looks like a bug on my part, will try to fix.

**Comment by pygy** [comment]
@threepointone [this](https://github.com/j2css/j2c/blob/5d381c2d721d04b54fabe6a165d587247c3087cb/src/helpers.js#L28-L61) may be handy...

Since it is slow, I search for a coma using `indexOf` and only run `splitSelector` when the simple test succeeds.

**Comment by threepointone** [comment]
@pygy that looks super helpful, thank you very much!

**Comment by pygy** [comment]
Glad you like it!

Please note that it requires a well-formed selectors... An extra paren (of either kind) may cause a silent,  incorrect split.

**Comment by jeroenransijn** [comment]
What is the status on this?

**Comment by threepointone** [comment]
been a while coming, but I just landed `:matches` support, live in 2.20.33
